e2e/util/wait.go

```diff
@@ -0,0 +1,142 @@
+// SPDX-FileCopyrightText: The RamenDR authors
+// SPDX-License-Identifier: Apache-2.0
+
+package util
+
+import (
+	"fmt"
+	"reflect"
+	"time"
+
+	argocdv1alpha1hack "github.com/ramendr/ramen/e2e/argocd"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8stypes "k8s.io/apimachinery/pkg/types"
+	ocmv1b1 "open-cluster-management.io/api/cluster/v1beta1"
+	ocmv1b2 "open-cluster-management.io/api/cluster/v1beta2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/ramendr/ramen/e2e/types"
+)
+
+func WaitForApplicationSetDelete(
+	ctx types.Context,
+	cluster types.Cluster,
+	name, namespace string,
+	deadline time.Time,
+) error {
+	applicationSetObj := &argocdv1alpha1hack.ApplicationSet{
```

Comment:
obj is a good name in this context. This is a tiny function with one variable.

Same for the other helpers.